Aarch64 imx8 crosscompile kvms

grihey opened the pull request:
###### Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

grihey committed:
Kernel patch for kvms support. Taken from kvms public repo.

Signed-off-by: Grigoriy Romanov <grigoriy.romanov@unikie.com>

grihey committed:
Uboot patch for load kvms automatically. Patch 0001-imx8qm-uefi.patch
was removed hence that changes already in imx8-uboot repo.

Signed-off-by: Grigoriy Romanov <grigoriy.romanov@unikie.com>

grihey committed:
Specific kernel with an applying of the kvms patch was added.

Signed-off-by: Grigoriy Romanov <grigoriy.romanov@unikie.com>

grihey committed:
Kvms package presense. Package fetch prebuild kvms binary images
from my personal github account. There are binary images for
imx8qxp and imx8qm boards.

Signed-off-by: Grigoriy Romanov <grigoriy.romanov@unikie.com>

vilvo commented on pkgs/os-specific/linux/kernel/0001-KVM-external-hypervisor-for-imx8-5.10.72-kernel.patch:
```diff
@@ -0,0 +1,2051 @@
+From e83aaa5788a779cae1e28f976a624647ee2fa0fa Mon Sep 17 00:00:00 2001
+From: Janne Karhunen <Janne.Karhunen@gmail.com>
+Date: Tue, 21 Apr 2020 09:45:23 +0300
+Subject: [PATCH] KVM: external hypervisor for imx8qmmek 5.10.72 kernel
```
Does this apply to the newer `5.15.32` kernel we updated yesterday? https://github.com/tiiuae/nixpkgs-spectrum/commit/60ddbbe6ccdebd1ec7f8bbc60cf2c2ead0824248

vilvo commented on pkgs/misc/uboot/default.nix:
```diff
@@ -553,7 +556,7 @@ in {
       ref = "lf_v2022.04-uefi";
     };
     patches = [
-      ./0001-imx8qm-uefi.patch
```
please sync with @panufi on if just replacing is even an option

vilvo commented on pkgs/os-specific/linux/kvms/default.nix:
```diff
@@ -0,0 +1,13 @@
+stdenv.mkDerivation ({
+
+    name = "kvms";
+
```
Please add comment on how and why binary blob is built from https://github.com/jkrh/kvms

vilvo reviewed (changes requested):
Please address comments. Also, testing needed. Thanks for the work so far - development builds can be built from your forks.

grihey committed:
KVMS kernel patch for nxp's lf-5.15.y branch baseline.

nxp's repo sha: fa6c3168

Signed-off-by: Grigoriy Romanov <grigoriy.romanov@unikie.com>

grihey committed:
KVMS patch for Uboot.

In a case when kvms binary is not presented on boot partition
uboot will just notify about that and will continue to load
device tree blob and linux image. lack of bl1.bin file dont
block boot process.

Signed-off-by: Grigoriy Romanov <grigoriy.romanov@unikie.com>

grihey committed:
KVMS support for 5.15.32 nxp's linux kernel

* Linux kernel patch for 5.15.32
* Kernel patch file for 5.10.72 was renamed
* kvms binaries for 5.15.32 kernel were added to external github
  repo, related package updated
* tiny clean up in uboot files

Signed-off-by: Grigoriy Romanov <grigoriy.romanov@unikie.com>

grihey committed:
KVMS patch for 5.15.32 kernel - vm launch doesn't fails

When kvms for 5.15 kernel take place, vCPU has a rised
KVM_ARM_VCPU_PMU_V3 bit in features description during
its creation. Te fact leads to return from kvm_vcpu_ioctl()
handler with error code -EINVAL.

Problem was resolved by forcely make pmu presence check
in vCPU to zero (lack of pmu). its hackish, but works.

Signed-off-by: Grigoriy Romanov <grigoriy.romanov@unikie.com>

grihey committed:
KVMS package got updated commit and sha256

Signed-off-by: Grigoriy Romanov <grigoriy.romanov@unikie.com>
